tools/expotools/src/commands/ClientBuild.ts

```diff
@@ -1,144 +1,177 @@
 import { Command } from '@expo/commander';
-import spawnAsync from '@expo/spawn-async';
-import { Config, UpdateVersions } from '@expo/xdl';
 import aws from 'aws-sdk';
 import chalk from 'chalk';
 import fs from 'fs-extra';
 import inquirer from 'inquirer';
 import path from 'path';
 
-import { STAGING_API_HOST, EXPO_DIR, IOS_DIR, ANDROID_DIR } from '../Constants';
-import { Platform, iosAppVersionAsync, androidAppVersionAsync } from '../ProjectVersions';
+import { EXPO_DIR } from '../Constants';
 import askForPlatformAsync from '../utils/askForPlatformAsync';
+import askForSdkVersionAsync from '../utils/askForSDKVersionAsync';
+import { link } from '../Formatter';
 import Git from '../Git';
+import logger from '../Logger';
+import { modifySdkVersionsAsync, getSdkVersionsAsync } from '../Versions';
+import { ClientBuilder, Platform } from '../client-build/types';
+import IosClientBuilder from '../client-build/IosClientBuilder';
+import AndroidClientBuilder from '../client-build/AndroidClientBuilder';
+
+const s3Client = new aws.S3({ region: 'us-east-1' });
+const { yellow, blue, magenta } = chalk;
 
 type ActionOptions = {
-  platform?: 'ios';
+  platform?: Platform;
   release: boolean;
+  skipUpload: boolean;
+};
+
+export default (program: Command) => {
+  program
+    .command('client-build')
+    .alias('cb')
+    .description(
+      'Builds Expo client for iOS simulator or APK for Android, uploads the archive to S3 and saves its url to versions endpoint.'
+    )
+    .option('-p, --platform [string]', 'Platform for which the client will be built.')
+    .option(
+      '-r, --release',
+      'Whether to upload and release the client build to staging versions endpoint.',
+      false
+    )
+    .option(
+      '--skip-upload',
+      "Whether to skip the uploading part. Might be useful for debugging and not to unintentionally override production's build.",
+      false
+    )
+    .asyncAction(main);
 };
 
-async function askToRecreateSimulatorBuildAsync(archivePath: string) {
+async function main(options: ActionOptions) {
+  const platform = options.platform || (await askForPlatformAsync());
+  const sdkBranchVersion = (await Git.getSDKVersionFromBranchNameAsync()) || '39.0.0';
```

Comment:
```suggestion
  const sdkBranchVersion = (await Git.getSDKVersionFromBranchNameAsync());
```
I dislike this hardcoded version here. Why not to add some prompt for entering a fallback version?
That is causing another point to pay attention to when dealing with the release process.

Reply:
tbh it's my mistake, I've added this fallback just for testing, I'll remove it.